src/xo-app/settings/remotes/index.js

```diff
@@ -0,0 +1,313 @@
+import _, { messages } from 'messages'
+import ActionButton from 'action-button'
+import filter from 'lodash/filter'
+import Icon from 'icon'
+import isEmpty from 'lodash/isEmpty'
+import map from 'lodash/map'
+import React, { Component } from 'react'
+import { connectStore } from 'utils'
+import { error } from 'notification'
+import { injectIntl } from 'react-intl'
+import {
+  createRemote,
+  deleteRemote,
+  disableRemote,
+  enableRemote,
+  subscribeRemotes
+} from 'xo'
+import {format, parse} from 'xo-remote-parser'
+
+const remoteTypes = {
+  local: 'remoteTypeLocal',
+  nfs: 'remoteTypeNfs',
+  smb: 'remoteTypeSmb'
+}
+
+class AbstractRemote extends Component {
+  render () {
+    const {
+      remote
+    } = this.props
+
+    return <tr>
+      <td></td>
+      <td>{remote.name}</td>
+      <td>{this._renderRemoteInfo(remote)}</td>
+      <td>{this._renderAuthInfo(remote)}</td>
+      <td>
+      {remote.enabled &&
+        <span>
+          <span className='text-success'>{this.accessible} <Icon icon='success' /></span>
+          {' '}
+          <ActionButton btnStyle='warning' handler={disableRemote} handlerParam={remote.id} icon='disconnect' />
+        </span>
+      }
+      {!remote.enabled &&
+        <span>
+          <span className='text-muted'>{this.unaccessible}</span>
+          {' '}
+          <ActionButton btnStyle='primary' handler={enableRemote} handlerParam={remote.id} icon='connect' />
+        </span>
+      }
+      </td>
+      <td><span className='text-muted'>{remote.error}</span></td>
+      <td><ActionButton btnStyle='danger' handler={deleteRemote} handlerParam={remote.id} icon='delete' /></td>
+    </tr>
+  }
+
+  _renderRemoteInfo (remote) {
+    throw new Error('NOT IMPLEMENTED')
+  }
+
+  _renderAuthInfo (remote) {
+    throw new Error('NOT IMPLEMENTED')
+  }
+
+  get accessible () {
+    throw new Error('NOT IMPLEMENTED')
+  }
+
+  get unaccessible () {
+    throw new Error('NOT IMPLEMENTED')
+  }
+}
+
+class LocalRemote extends AbstractRemote {
+  _renderRemoteInfo (remote) {
+    return remote.path
+  }
+
+  _renderAuthInfo (remote) {
+    return ''
+  }
+
+  get accessible () {
+    return 'Accessible'
+  }
+
+  get unaccessible () {
+    return 'Unaccessible'
+  }
+}
+
+class NfsRemote extends AbstractRemote {
+  _renderRemoteInfo (remote) {
+    return `${remote.host}:${remote.share}`
+  }
+
+  _renderAuthInfo (remote) {
+    return ''
+  }
+
+  get accessible () {
+    return 'Mounted'
+  }
+
+  get unaccessible () {
+    return 'Unmounted'
+  }
+}
+
+class SmbRemote extends AbstractRemote {
+  _renderRemoteInfo (remote) {
+    return <span><strong className='text-info'>\\</strong>{remote.host}<strong className='text-info'>\</strong>{remote.path}</span>
+  }
+
+  _renderAuthInfo (remote) {
+    return <span>{remote.username}@{remote.domain}</span>
+  }
+
+  get accessible () {
+    return 'Accessible'
+  }
+
+  get unaccessible () {
+    return 'Unaccessible'
+  }
+}
+
+@injectIntl
```

Comment:
It seems I can't passa function a second parameter of _ like the example you linked...